bin/alpen-bridge/src/rpc_server.rs

```diff
@@ -685,47 +714,22 @@ impl StrataBridgeMonitoringApiServer for BridgeRpc {
 
     async fn get_claims(&self) -> RpcResult<Vec<Txid>> {
         // Use the cached contracts
-        let all_entries = self.cached_contracts.read().await.clone();
+        let cache = self.cached_contracts.read().await.clone();
 
-        let claims: Vec<_> = all_entries
-            .into_iter()
-            .filter_map(|entry| match entry.0.state.state {
-                // States that have an active graph with a claim transaction.
-                ContractState::Claimed { active_graph, .. }
-                | ContractState::Challenged { active_graph, .. }
-                | ContractState::PreAssertConfirmed { active_graph, .. }
-                | ContractState::AssertDataConfirmed { active_graph, .. }
-                | ContractState::Asserted { active_graph, .. }
-                | ContractState::Fulfilled { active_graph, .. } => Some(active_graph.1.claim_txid),
-
-                // States that do not have an active graph with a claim transaction.
-                ContractState::Requested { .. }
-                | ContractState::Deposited { .. }
-                | ContractState::Assigned { .. } => None,
-
-                // States that are terminal and do not have an active graph.
-                ContractState::Disproved { .. } | ContractState::Aborted => None,
-
-                // However resolved contracts have a definite `claim_txid`.
-                ContractState::Resolved { claim_txid, .. } => Some(claim_txid),
-            })
+        let claims = Self::get_contracts_with_claims(&cache)
+            .map(|(_, claim_txid)| claim_txid)
             .collect();
 
         Ok(claims)
     }
 
     async fn get_claim_info(&self, claim_txid: Txid) -> RpcResult<Option<RpcClaimInfo>> {
         // Use the cached contracts
-        let all_entries = self.cached_contracts.read().await.clone();
-
-        let claim_info = all_entries
-            .iter()
-            .find(|entry| {
-                let claim_txids = entry.0.state.state.claim_txids();
+        let cache = self.cached_contracts.read().await.clone();
 
-                claim_txids.contains(&claim_txid)
-            })
-            .map(|entry| contract_state_to_reimbursement_status(&entry.0.state.state))
+        let claim_info = Self::get_contracts_with_claims(&cache)
+            .find(|(_, entry_claim_txid)| *entry_claim_txid == claim_txid)
+            .map(|(entry, _)| contract_state_to_reimbursement_status(&entry.0.state.state))
```

Comment:
by the functor composition law: `map g . map f == map (g . f)`

Reply:
Right! Amended in fc78f9b51e347b27576268060c630ae5f4ec0103